src/Experimental/Experimental.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * Experimental features for WPGraphQL, hidden behind feature flags.
+ *
+ * @package WPGraphQL\Experimental
+ */
+
+namespace WPGraphQL\Experimental;
+
+/**
+ * Class - Experimental
+ */
+final class Experimental {
+	/**
+	 * Whether Experimental Functionality is enabled or not.
+	 *
+	 * @var ?bool
+	 */
+	protected $is_enabled;
+
+	/**
+	 * The Experiment Registry instance
+	 *
+	 * @var \WPGraphQL\Experimental\ExperimentRegistry
+	 */
+	protected $experiment_registry;
+
+	/**
+	 * Initializes Experimental Functionality for WPGraphQL
+	 */
+	public function init(): void {
+		if ( ! $this->is_enabled() ) {
+			return;
+		}
+
+		// Initialize the Experiment registry.
+		$this->experiment_registry = new ExperimentRegistry();
+		$this->experiment_registry->init();
+
+		// Register Admin functionality.
+		if ( is_admin() ) {
+			$admin = new Admin();
+			$admin->init();
+		}
+	}
+
+	/**
+	 * Whether Experimental Functionality is enabled or not.
+	 */
+	protected function is_enabled(): bool {
+		if ( ! isset( $this->is_enabled ) ) {
+			$this->is_enabled = defined( 'GRAPHQL_EXPERIMENTAL_FEATURES' ) ? (bool) GRAPHQL_EXPERIMENTAL_FEATURES : true;
```

Comment:
Is there any benefit in adding a `filter hook` option here?

Reply:
Here specifically I'm not sure about, since the  intent of the check here is to allow a user to intentionally disable the _entire Experimental module_.

We might want [a filter in `AbstractExperiment`](https://github.com/wp-graphql/wp-graphql/blob/d51f9c09fd1beccfb0d96d56a68efe48cd20b6fb/src/Experimental/Experiment/AbstractExperiment.php#L121) beyond what `graphql_get_setting()` provides in order to allow for codependent Experiments without overloading the entire `is_active()` method 🧐

Reply:
Added `wp_graphql_experiment_enabled` in d125d5088ea07297aa39d8e537d674bfb3357056. I believe that solves the use case for this.